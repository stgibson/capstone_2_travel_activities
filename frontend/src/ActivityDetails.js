import React, { useState, useEffect } from "react";
import { useParams } from "react-router-dom";

const ActivityDetails = ({ getActivityDetails }) => {
  const { id } = useParams();
  const [currActivity, setCurrActivity] = useState(null);

  useEffect(() => {
    const getActivity = async () => {
      const activity = await getActivityDetails(id);
      if (activity) {
        setCurrActivity(activity);
      }
    };
    getActivity();
  }, [id, getActivityDetails])

  return (
    <>
    {
      currActivity ? (
        <>
          <h1>{ currActivity.name }</h1>
          <div>{ currActivity.description }</div>
          <div>Rating: { currActivity.rating }</div>
          <div>Booking Link: { currActivity.bookingLink }</div>
          <div>Price: { currActivity.price }</div>
          <div>Currency: { currActivity.currencyCode }</div>
        </>
      ) : (
        <h1>...Loading</h1>
      )
    }
    </>
  );
};

export default ActivityDetails;