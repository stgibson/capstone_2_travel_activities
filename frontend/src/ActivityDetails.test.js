import React from "react";
import { MemoryRouter, Route } from "react-router-dom";
import { render } from "@testing-library/react";
import ActivityDetails from "./ActivityDetails";

const activity = {
  id: 1,
  name: "Test Activity",
  description: "Test Description",
  rating: "Test Rating",
  bookingLink: "Test Booking Link",
  price: "Test Price",
  currencyCode: "Test Currency Code"
};

let getActivityDetails;

beforeEach(() => {
  getActivityDetails = jest.fn(id => {
    if (id === 1) {
      return activity;
    }
    return null;
  });
});

// determined how to test components that use useParams at https://tomalexhughes.com/blog/testing-components-that-use-react-router-hooks
it("renders without crashing", () => {
  render(
    <MemoryRouter initialEntries={ ["/activities/1"] }>
      <Route path="/activities/:id">
        <ActivityDetails getActivityDetails={ getActivityDetails } />
      </Route>
    </MemoryRouter>
  );
});

it("renders without crashing", () => {
  const { asFragment } = render(
    <MemoryRouter initialEntries={ ["/activities/1"] }>
      <Route path="/activities/:id">
        <ActivityDetails getActivityDetails={ getActivityDetails } />
      </Route>
    </MemoryRouter>
  );

  expect(asFragment()).toMatchSnapshot();
});