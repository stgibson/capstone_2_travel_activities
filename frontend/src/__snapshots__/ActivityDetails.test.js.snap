// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<DocumentFragment>
  <h1>
    ...Loading
  </h1>
</DocumentFragment>
`;
